File: python/04_balance.py
"""How many credits are left, and how much of the window is still free.

This endpoint sits beside the API rather than under /v1, so it is a plain HTTP
call rather than an SDK method.

    python 04_balance.py
"""

import json
import os
import urllib.request

# Naming your client in the user agent is good manners and makes your own
# requests easy to spot in a log.
request = urllib.request.Request(
    "https://onomeo.com/api/me",
    headers={
        "Authorization": f"Bearer {os.environ['ONOMEO_API_KEY']}",
        "User-Agent": "onomeo-examples/1.0",
    },
)

with urllib.request.urlopen(request) as response:
    me = json.load(response)

checkin = me["checkin"]
calls = me["calls"]
hours = round(calls["windowMinutes"] / 60)

print(f"balance        {me['balance']:,} credits")
print(f"spent so far   {me['spentTotal']:,} credits")

if checkin["claimedToday"]:
    print(f"check-in       claimed; tomorrow is worth {checkin['nextAmount']:,}")
else:
    print(f"check-in       {checkin['todayAmount']:,} credits waiting on the dashboard")

print(f"calls left     {calls['left']} of {calls['max']} in this {hours}-hour window")

File: javascript/04-balance.mjs
/* How many credits are left, and how much of the window is still free.
 *
 * This endpoint sits beside the API rather than under /v1, so it is a plain
 * fetch rather than an SDK method.
 *
 *   node 04-balance.mjs
 */

const response = await fetch('https://onomeo.com/api/me', {
  headers: { Authorization: `Bearer ${process.env.ONOMEO_API_KEY}` },
});

if (!response.ok) {
  const { error } = await response.json();
  throw new Error(`${error.code}: ${error.message}`);
}

const me = await response.json();
const hours = Math.round(me.calls.windowMinutes / 60);
const n = (value) => value.toLocaleString('en-US');

console.log(`balance        ${n(me.balance)} credits`);
console.log(`spent so far   ${n(me.spentTotal)} credits`);
console.log(
  me.checkin.claimedToday
    ? `check-in       claimed; tomorrow is worth ${n(me.checkin.nextAmount)}`
    : `check-in       ${n(me.checkin.todayAmount)} credits waiting on the dashboard`,
);
console.log(`calls left     ${me.calls.left} of ${me.calls.max} in this ${hours}-hour window`);
